doc/answerfile.txt

```diff
@@ -36,7 +37,7 @@ Restore:
 Common Elements
 ---------------
 
-  <source type="type">repo</source>
+  <source type="type">repo</source>+
```

Comment:
`<source>` is not required for the `<restore>` mode so I think this should be `*`.

Reply:
Hm, that makes things harder to document in a readable way :)
A `*` and an additional note could possibly work.

I'm not sure how much it would help people on the documenting side, but I had the thought of adding a DTD (old-school that I am...), which could at least help to validate an answerfile.  But then, with `<installation mode="upgrade">` not accepting the same content as the other `<installation>` modes, we would not be able to write a DTD that would reject all invalid files - we would need to use the additional expressiveness of an XML Schema.  Does that sound like a useful addition ?

While we're at that, I was also wondering how much of a good idea it to be to consider using a separate `<upgrade>` tag.  It seems different enough to justify it, and would probably make the doc easier to write and read, but if we want backward compatibility the implementation could suffer a bit in readability.

Which also brings me to the question of deprecation of old syntax: there are already some features documented as deprecated, and I did not see any explicit timeframe for their removal.  All of this should probably be separate discussions, do you think it's worth opening tickets to followup on them ?